src/docs/arch-storage-extension.dox

```diff
@@ -0,0 +1,104 @@
+/*! @arch_page arch-cloud-storage-extension Storage Source
+
+@section extension_storage_source Cloud Storage Source Implementation
+
+The WiredTiger extension allows the control of how files are stored via the storage source
```

Comment:
Slightly weird grammar here: "The WiredTiger extension allows the control" -> "The WiredTiger extension API allows control".

Reply:
done!